src/reducers/filterReducer.js

```diff
@@ -0,0 +1,61 @@
+const filterActions = {
+    FILTER_BY_GENRE: 'FILTER_BY_GENRE',
+    FILTER_BY_PRICE: 'FILTER_BY_PRICE',
+    FILTER_BY_RATINGS: 'FILTER_BY_RATINGS',
+    FILTER_BY_BOOK_TYPE: 'FILTER_BY_BOOK_TYPE',
+    SORT_BY: 'SORT_BY',
+    CLEAR_FILTERS: 'CLEAR_FILTERS'
+};
+
+const initialFilterState = {
+    sortBy: '',
+    genres: {
```

Comment:
You can think of getting this genres data from api (if possible) and populate it in the state. Again this looks great but just a suggestion.

Reply:
Yes, I am planning to do this when I add my very own backed to the app.